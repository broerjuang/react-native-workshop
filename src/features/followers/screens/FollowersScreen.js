// @flow

import React, {Component} from 'react';
import {ScrollView, View, TouchableOpacity, Text} from 'react-native';
import {SearchBar} from 'react-native-elements';
import Icon from '../../../global/core-ui/Icon';
import {connect} from 'react-redux';
import {ListItem} from 'react-native-elements';
import fetchJSON from '../../../global/helpers/fetchJSON';

import type {Repo} from '../types';

type Props = {
  // total_count: number;
  // incomplete_results: boolean;
  // items: Array<Items>;
  navigation: Object;
  handleAction: (action: Object) => void;
  followersData: Array<*>;
};
type State = {
  items: Array<Repo>;
  search: string;
  followersList: Array<*>;
};
export class FollowersScreen extends Component<Props, State> {
  static navigationOptions = (options: *) => ({
    headerLeft: (
      <View style={{paddingLeft: 10}}>
        <TouchableOpacity
          onPress={() => {
            options.navigation.goBack(null);
          }}
        >
          <Icon
            name={'arrow-back'}
            size={30}
            color={'black'}
            type={'MATERIAL_ICONS'}
          />
        </TouchableOpacity>
      </View>
    ),
    headerTitle: (
      <Text style={{fontSize: 24, fontWeight: 'bold'}}>Followers</Text>
    ),
  });
  state = {
    items: [],
    search: '',
    followersList: [],
  };

  componentDidMount() {
    this.props.handleAction({type: 'FOLLOWERS_REQUESTED'});
  }

  render() {
    let {items, search} = this.state;
    items = this.props.followersData;
    let showItems = [];
    if (search !== '') {
      items.forEach((item) => {
        if (item.login.toLowerCase().includes(search.toLowerCase())) {
          showItems.push(item);
        }
      });
    } else {
      showItems = this.props.followersData;
    }

    return (
      <View style={{flex: 1}}>
        <SearchBar
          onChangeText={(text: string): void => this.setState({search: text})}
          onClearText={() => this.setState({search: ''})}
          value={this.state.search}
          placeholderTextColor={'#E1E4EC'}
          placeholder={'Type Here...'}
          icon={{
            type: 'material',
            color: '#909AA3',
            name: 'search',
            style: {alignContent: 'center'},
          }}
          inputStyle={{backgroundColor: '#E1E4EC'}}
          containerStyle={{
            backgroundColor: 'white',
            borderBottomWidth: 0.1,
            borderTopWidth: 0.5,
          }}
        />
        <ScrollView>
          {showItems.map((item) => (
            <ListItem
              roundAvatar
              avatar={{uri: item.avatar_url}}
              key={item.id}
              title={item.login}
              onPress={() =>
                this.props.navigation.navigate('Profile', {
                  userLogin: item.login,
                  userFullName: item.name,
                  userPicture: item.avatar_url,
                  sumRepositories: item.public_repos,
                  sumFollowers: item.followers,
                  sumFollowing: item.following,
                  sumStars: 0,
                  biography: item.bio,
                  website: item.blog,
                })
              }
            />
          ))}
        </ScrollView>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    userName: state.loginReducer.userName,
    followersData: state.followersReducer.followersData,
  };
}
function mapDispatchToProps(dispatch) {
  return {
    handleAction: (action: Object) => dispatch(action),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(FollowersScreen);
