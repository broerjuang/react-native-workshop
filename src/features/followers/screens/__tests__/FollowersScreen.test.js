// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import {FollowersScreen} from '../FollowersScreen';

describe('Followers Screen', () => {
  it('should render Followers corectly', () => {
    let component = renderer.create(
      <FollowersScreen
        followersData={[]}
        handleAction={() => {
          return;
        }}
        navigation={{}}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
