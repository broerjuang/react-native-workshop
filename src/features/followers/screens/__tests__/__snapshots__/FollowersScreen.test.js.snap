// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Followers Screen should render Followers corectly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
        },
        false,
        Object {
          "backgroundColor": "white",
          "borderBottomWidth": 0.1,
          "borderTopWidth": 0.5,
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onClearText={[Function]}
      placeholder="Type Here..."
      placeholderTextColor="#E1E4EC"
      style={
        Array [
          Object {
            "backgroundColor": "#303337",
            "borderRadius": 3,
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 30,
            "margin": 8,
            "overflow": "hidden",
            "paddingLeft": 26,
            "paddingRight": 19,
          },
          false,
          false,
          false,
          Object {
            "backgroundColor": "#E1E4EC",
          },
          undefined,
          false,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#909AA3",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "position": "absolute",
              "top": 15.5,
            },
            Object {
              "left": 16,
            },
            Object {
              "alignContent": "center",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î¢¶
    </Text>
  </View>
  <RCTScrollView>
    <View />
  </RCTScrollView>
</View>
`;
